utils/variables.go

```diff
@@ -8,14 +8,67 @@ import (
 	"github.com/pkg/errors"
 )
 
-// ValidNameRegex is the pattern that matches to a valid name.
-// The name must begin with a letter i.e. [a-zA-Z],
-// and the body can only contain 0 or more numbers, letters, dashes and underscores i.e. [-\w]*.
-var ValidNameRegex = regexp.MustCompile(`^[a-zA-Z][-\w]*$`)
-
-// ErrInvalidName returns a human-readable error for when ValidNameRegex doesn't match.
-func ErrInvalidName(name string) error {
-	return errors.Errorf("name %q must start with a letter and must only contain letters, numbers, dashes, and underscores", name)
+var (
+	// validResourcenameRegex is the pattern that matches to a valid name.
+	// The name must begin with a letter or number i.e. [a-zA-Z0-9],
+	// and contains only letters, numbers, dashes, and underscores i.e. [-\w]*.
+	//
+	// Note:
+	// This regex cannot be changed to allow for:
+	// - colons: used as delimiters for remote names
+	// - plus signs: used for WebRTC track names.
+	validResourceNameRegex       = regexp.MustCompile(`^[a-zA-Z0-9][-\w]*$`)
+	validResourceNameExplanation = "must start with a letter or number and must only contain letters, numbers, dashes, and underscores"
+
+	// the module name is used to create the socket path, so ensure that this only
+	// accepts socket paths.
+	validModuleNameRegex       = regexp.MustCompile(`^[a-zA-Z0-9][-\w]*$`)
+	validModuleNameExplanation = validResourceNameExplanation
+
+	validPackageNameRegex       = validModuleNameRegex
+	validPackageNameExplanation = validResourceNameExplanation
+)
+
+// ValidateResourceName validates that the resource follows our naming requirements.
+func ValidateResourceName(name string) error {
+	if len(name) > 60 {
```

Comment:
I'm doing the validation in code rather than in regex because it simplifies the regex and adds no additional code complexity.